### ISSUE TITLE
Feature Request: Custom Icons by State

### ISSUE THREAD

**Issue by julianrinaldi** [cgiesche/streamdeck-homeassistant#22]
First, this is an amazing plugin. It's literally the reason why I bought two Stream Decks. So far it works great. Is it possible to have a custom icon for binary switches like lights, switches and input booleans? Obviously, there are the icons that come with the plugin, so it's possible. If it's not possible through the Stream Deck app, is it possible to edit it some other way? Or is this a feature you could add?

**Comment by cgiesche** [comment]
Hi!

The word "icon" sounds very simple for what is going on in the code to generate them :). The stream deck api allows eihter to use exactly one or two predefined images that can be configured in the gui OR to use SVG-images.

Of course, i need more than two static images, so i am using SVG images. Sadly, i can not simply load svg images from files. I have to generate them in my code. Colors and text lines (except the tilte text) are part of the dynamic SVG templates. The "icon" you see above the Title is only a small part of the whole "button" image.

This is, how it looks like: https://github.com/cgiesche/streamdeck-homeassistant/blob/master/src/modules/plugin/imageUtils.js
Scroll down to take a look at hell :D

Currently I am using the same "icon" for different types of entities, but i could use different icons based on following logic:

`[Domain][device_class (optional)][state (optional)]`

`[switch][*][on]` -> on-switch-image
`[switch][*][off]` -> off-switch-image
`[light][*][on]` -> defaults to `[switch][*][on]` 
`[light][*][off]` -> defaults to `[switch][*][off]`
`[input_boolean][*][on]` -> defaults to `[switch][*][on]` 
`[input_boolean][*][off]` -> defaults to `[switch][*][off]`

`[sensor][battery][*]` -> battery icon with loading state
`[sensor][temperature][*]` -> thermometer icon
etc...

My main problem is: I am not good at making icons that are looking good, still looking good on the streamdeck button and looking uniformly.

Long story short: I could easily add different images for different types of entities (fan/light/switch/pressure sensor/...) but i did not find an iconset that doesn't look like a mess when scaled down to about 65 pixels in height.

Making icons customizable by the user will not happen in the near future :/
